src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/tasks/CleanupEnvelopesDlqTask.java

```diff
@@ -113,10 +110,14 @@ private void logMessage(IMessage msg) {
 
     private boolean canBeCompleted(IMessage message) {
         Instant cutoff = Instant.now().minus(this.ttl);
-        Map<String, String> messageProperties = message.getProperties();
+        Map<String, Object> messageProperties = message.getProperties();
+
+        Instant deadLetteredAt =
+            messageProperties == null
+            ? null
+            : (Instant) messageProperties.get("deadLetteredAt");
```

Comment:
why not `Instant.parse(messageProperties.get("deadLetteredAt"))`? (as before)

Reply:
now we can keep objects , not limited to string